src/discourse.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface for interactions with discourse."""
+import logging
+import typing
+
+from ops.pebble import ChangeError, ExecError, ExecProcess
+
+CONTAINER_APP_USERNAME = "_daemon_"
```

Comment:
I think we define this in `src/charm.py` currently, so can we either refer to that one here, or refer to this one there?

Reply:
I think the end goal is to have the constant defined here and refer to it in `charm.py`. But imo refactoring `charm.py` is quite a significant change, so I'd prefer to keep things simple in this PR, then work on the refactoring on another PR.

The idea is also to move most of the calls to `bundle exec rake ...` from `charm.py` to this module. So I think we can leave it like this for now, then if this is ok for @merkata and @nrobinaubertin, do a follow-up PR refactoring `charm.py`.

What do you think?

Reply:
Sure, if they're okay with it, sounds good to me too.